Jenkinsfile.integration

```diff
@@ -56,23 +58,24 @@ pipeline {
                     sh "git fetch --no-tags"
                     List<String> sourceChanged = sh(returnStdout: true, script: "git diff --name-only origin/master..origin/${env.BRANCH_NAME}").split()
                     echo "Files changed for this PR:\n${sourceChanged.join('\n')}"
-                    for (int i = 0; i < sourceChanged.size(); i++) {
-                        if (sourceChanged[i].contains("playbooks/")) {
-                            PlaybooksChanged = true
-                        }
-                        if (sourceChanged[i].contains("doc/")) {
-                            DocsChanged = true
-                        }
-                        if (sourceChanged[i].contains("tox.ini")) {
-                            ToxChanged = true
-                        }
-                    }
+                    PlaybooksChanged = sourceChanged.any{it.contains("doc/")}
+                    DocsChanged = sourceChanged.any{it.contains("playbooks/")}
+                    ToxChanged = sourceChanged.any{it.contains("tox.ini")}
+                    JenkinsfileChanged = sourceChanged.any{it.contains("jenkinsFile.integration")}
+
+                    DocsOnlyRun = DocsChanged && !PlaybooksChanged && !ToxChanged && !JenkinsfileChanged
```

Comment:
I think this creates falls positives. `DocsRunOnly` will e.g. be true if the submodule references are updated or if files in `files/` (e.g the velum automation tarball) are touched. 

Something like this might work though: 
`DocOnlyRun = sourceChanged.every{it.contains("doc/")}`

Reply:
nice!